src/login/logind-button.c

```diff
@@ -52,6 +54,33 @@ Button* button_new(Manager *m, const char *name) {
         return b;
 }
 
+static void* graceful_close_thread(void *p) {
+        (void) pthread_setname_np(pthread_self(), "close");
+        (void) close(PTR_TO_FD(p));
+        return NULL;
+}
+
+static int graceful_asynchronous_close(int fd) {
+        int r;
+
+        /* Apparently closing input devices is really slow (> 100ms), hence let's do so asynchronously. This
+         * is pretty similar to asynchronous_close(), but with one big difference: If the device has been
+         * unplugged close() returns ENODEV, let's ignore this. We don't use safe_close()/close_nointr()
+         * internally hence. */
+
+        if (fd >= 0) {
+                PROTECT_ERRNO;
+
+                r = asynchronous_job(graceful_close_thread, FD_TO_PTR(fd));
+                if (r < 0)
+                        (void) close_nointr(fd);
+        }
+
+        return -1;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(int, graceful_asynchronous_close);
```

Comment:
That part doesnt' seem necessary, because it does pretty much the same thing that `asynchronous_close` does.
(asynchronous_close calls safe_close which only cares that we don't get EBADF.)

Reply:
hmm, indeed. not sure what i was smoking. updated. ptal!